pkg/controller/controller.go

```diff
@@ -343,12 +343,18 @@ func (ctrl *resizeController) syncPVC(key string) error {
 		klog.V(4).Infof("No need to resize PVC %q", util.PVCKey(pvc))
 		return nil
 	}
-	if !ctrl.pvNeedResize(pvc, pv) {
-		klog.V(4).Infof("No need to resize PV %q", pv.Name)
-		return nil
-	}
 
-	return ctrl.resizePVC(pvc, pv)
+	if utilfeature.DefaultFeatureGate.Enabled(features.RecoverVolumeExpansionFailure) {
+		_, _, err, _ := ctrl.expandAndRecover(pvc, pv)
+		return err
+	} else {
+		if !ctrl.pvNeedResize(pvc, pv) {
```

Comment:
seem pvNeedResize can be called before checking feature is enabled?

Reply:
is pvNeedResize more like pv cannot be resized due to condition is not met? Does it consider an error?

Reply:
this is basically old flow - which checks if PV is resizable (such as, whether it is bound and bound to PVC which is being resized and whether it was already controller expanded and node expansion is pending). It is not a hard error if PV can not expanded at this moment. But anyways - this is old code and has kind always worked okay.

Reply:
oh I confused about pvCanBeExpanded and pvNeedResize, thinking they are the same function.